scripts/pre_commit_linter.py

```diff
@@ -881,8 +883,91 @@ def _check_html_directive_name(all_files):
+    print '----------------------------------------'
+    summary_messages = []
+    files_to_check = [
+        filename for filename in all_files if not
+        any(fnmatch.fnmatch(filename, pattern) for pattern in EXCLUDED_PATHS)
+        and filename.endswith('.js')]
+    failed = False
+    directive_has_isolated_scope = False
+    summary_messages = []
+    parser = pyjsparser.PyJsParser()
+    for filename in files_to_check:
+        with open(filename) as f:
+            content = f.read()
+        parsed_nodes = parser.parse(content)['body']
+        for parsed_node in parsed_nodes:
+            if parsed_node['type'] != 'ExpressionStatement':
+                continue
+            expression = parsed_node['expression']
+            if not (expression['type'] == 'CallExpression' and (
+                    expression['callee']['type'] == (
+                        'MemberExpression')) and (
+                            expression['callee']['property']['name'] == (
+                                'directive'))):
+                continue
+            arguments = expression['arguments']
+            directive_name = str(arguments[0]['value']) if (
+                arguments[0]['type'] == 'Literal') else (
+                    "No such directive exists.")
+            arguments = arguments[1:]
+            for argument in arguments:
+                if argument['type'] != 'ArrayExpression':
+                    continue
+                elements = argument['elements']
+                for element in elements:
+                    if element['type'] != 'FunctionExpression':
+                        continue
+                    body = element['body']
+                    if body['type'] != 'BlockStatement':
+                        continue
+                    body_elements = body['body']
+                    for body_element in body_elements:
+                        if not (body_element['type'] == 'ReturnStatement' and (
+                                body_element['argument']['type'] == 'ObjectExpression')):
+                            continue
+                        return_node_properties = body_element['argument']['properties']
+                        for return_node_property in return_node_properties:
+                            if return_node_property['key']['type'] == 'Identifier' and (
+                                return_node_property['key']['name'] == 'scope'):
+                                scope_value = return_node_property['value']
+                                if scope_value['type'] == 'ObjectExpression':
+                                    directive_has_isolated_scope = True
+                                elif scope_value['type'] == 'Literal' and scope_value['value']:
+                                    failed = True
+                                    print 'Please ensure that %s directive in %s file does not have scope set to true.' %(directive_name, filename)
```

Comment:
Why is this really long line not caught by the linter?

Add space between % and (

Reply:
Ah. Is it worth not excluding it? I'm not sure I see the point of exempting this script from lint checks...